src/Symfony/Component/Yaml/Inline.php

```diff
@@ -203,6 +203,13 @@ public static function dump($value, $flags = 0)
                 return "''";
             case Yaml::DUMP_BASE64_BINARY_DATA & $flags && self::isBinaryString($value):
                 return '!!binary '.base64_encode($value);
+        }
+
+        if (self::isBinaryString($value)) {
+            @trigger_error('Dumping non UTF-8 data without passing the DUMP_BASE64_BINARY_DATA flag is deprecated since Symfony 3.1 and will be removed in 4.0.', E_USER_DEPRECATED);
```

Comment:
I'm wondering if `Yaml::DUMP_BASE64_BINARY_DATA` should not be set by default in 4.0 and the option deprecated. that way, the YAML dumper always behaves as expected without having to change some options which seems weird.